packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -571,6 +574,20 @@ export class Function extends FunctionBase {
     });
   }
 
+  /**
+   * The LogGroup where the Lambda function's logs are made available.
+   */
+  public get logGroup(): logs.ILogGroup {
+    if (!this._logGroup) {
+      const logretention = new LogRetention(this, 'LogRetention', {
+        logGroupName: `/aws/lambda/${this.functionName}`,
+        retention: logs.RetentionDays.INFINITE,
+      });
```

Comment:
Do we even _need_ this? Can't we just `fromAttributes` the LogGroup?

I feel like setting `INFINITE` retention here may not be the right thing to do... People may have manually changed the retention to something else before they went through this code path, and they probably don't expect you'll be resetting it to `Expire: NEVER` when they just access the `logGroup` property...

Now obviously, you could have the problem that the LogGroup will not exist in the service before the function has been invoked at least once (if you don't create it yourself)... In such case, it'd be better to have a custom handler that just creates the log group if missing, but leaves the configured retention if it already exists...

Reply:
> Now obviously, you could have the problem that the LogGroup will not exist in the service before the function has been invoked at least once (if you don't create it yourself)... In such case, it'd be better to have a custom handler that just creates the log group if missing, but leaves the configured retention if it already exists...

LogRetention does exactly that.

> they probably don't expect you'll be resetting it to Expire: NEVER when they just access the logGroup property...

Good point. Let me look into that.

Reply:
>> they probably don't expect you'll be resetting it to Expire: NEVER when they just access the logGroup property...
>
> Good point. Let me look into that.

Wait a second. They shouldn't have set the retention policy of the function out of band; they should use the `logRetention` property here. If they did set it out of band, it's expected that a stack deployment may reset it, right?

Reply:
That's not what Vanilla CloudFormation would do for you... I don't have a problem if you'll be resetting aggressively and consider you have to change it through here, but this should be mentioned in the documentation at the very least!

Reply:
Maybe the correct solution is to make the retention property optional and have the custom resource do nothing if the property is unset and the group already exists? 

On the other hand, I don't think this is going to come up a lot. Who really precreates Lambda Function Log Groups AND subsequently uses CDK to manage their Lambdas? With generated names? Plus, the reset will only happen on the first deployment, because afterwards none of the resource properties will have changed so CFN is never going to touch this custom resource ever again.

Reply:
My concern is more towards the fact that a getter should be side-effects free, as it's only observing state. There are possible side-effects here, which isn't something you'd expect (and hence check for).

If your lambda logs stuff out that you are _not allowed_ to keep for longer than X for some reason... You may suddenly be in violation of law without knowing it...

An alternative way to go here would be to _always_ create a (regular) LogGroup, and use an auto-generated physical name (instead of ever leaving it out to CloudFormation to decide).

Reply:
> An alternative way to go here would be to always create a (regular) LogGroup, and use an auto-generated physical name (instead of ever leaving it out to CloudFormation to decide).

I'm not sure I follow your alternative here. 
However, reminding you that the lambda service decides that the log group must be `/aws/lambda/<function-name>` and we have no control over it.

> If your lambda logs stuff out that you are not allowed to keep for longer than X for some reason... You may suddenly be in violation of law without knowing it...

You're right, but if you need to comply with the law, you should be setting the `logRetention` property which will do the correct thing for you. By not managing an AWS resource that is regulated lawfully in you IaaC is already bad.

However, your philosophical point is well taken.

I've updated the README with more documentation on this. Does that satisfy, or do you feel strongly that I change the logic?